app/setup.py

```diff
@@ -0,0 +1,266 @@
+
+
+def add_blueprints(application):
+    if application.config['EQ_DEV_MODE']:
+        # import and register the dev mode blueprint
+        from app.views.dev_mode import dev_mode_blueprint
+        application.register_blueprint(dev_mode_blueprint)
+        application.debug = True
+
+    # import and register the main application blueprint
+    from app.views.questionnaire import questionnaire_blueprint
+    application.register_blueprint(questionnaire_blueprint)
+    questionnaire_blueprint.config = application.config.copy()
+
+    from app.views.session import session_blueprint
+    application.register_blueprint(session_blueprint)
+    session_blueprint.config = application.config.copy()
+
+    from app.views.flush import flush_blueprint
+    application.register_blueprint(flush_blueprint)
+    flush_blueprint.config = application.config.copy()
+
+    from app.views.dump import dump_blueprint
+    application.register_blueprint(dump_blueprint)
+    dump_blueprint.config = application.config.copy()
+
+    from app.views.errors import errors_blueprint
+    application.register_blueprint(errors_blueprint)
+    errors_blueprint.config = application.config.copy()
+
+    from app.jinja_filters import blueprint as filter_blueprint
+    application.register_blueprint(filter_blueprint)
+
+
+def setup_secure_cookies(application):
+    session_timeout = application.config['EQ_SESSION_TIMEOUT_SECONDS']
+    grace_period = application.config['EQ_SESSION_TIMEOUT_GRACE_PERIOD_SECONDS']
+    application.secret_key = application.config['EQ_SECRET_KEY']
+    application.permanent_session_lifetime = timedelta(seconds=session_timeout + grace_period)
+    application.session_interface = SHA256SecureCookieSessionInterface()
+    application.config['SESSION_COOKIE_SECURE'] = True
+
+
+def setup_babel(application):
+    application.babel = Babel(application)
+    application.jinja_env.add_extension('jinja2.ext.i18n')
+
+
+def add_safe_health_check(application):
+    @application.route('/status')
+    def safe_health_check():  # pylint: disable=unused-variable
+        data = {
+            'status': 'OK',
+            'version': application.config['EQ_APPLICATION_VERSION'],
+        }
+        return json.dumps(data)
+
+
+def versioned_url_for(endpoint, **values):
+
+    if endpoint == 'static':
+        filename = values.get('filename', None)
+        if filename:
+            filename = get_minimized_asset(filename)
+            # use the git revision
+            version = settings.EQ_APPLICATION_VERSION
+            values['filename'] = filename
+            values['q'] = version
+    return url_for(endpoint, **values)
+
+
+def get_minimized_asset(filename):
+    """
+    If we're in production and it's a js or css file, return the minified version.
+    :param filename: the original filename
+    :return: the new file name will be .min.css or .min.js
+    """
+    if settings.EQ_MINIMIZE_ASSETS:
+        if 'css' in filename:
```

Comment:
Probably, but EQ_MINIMIZE_ASSETS doesn't actually work anyway. and isn't really the context of this change.